boards/arm/quick_feather/doc/index.rst

```diff
@@ -0,0 +1,124 @@
+.. _quickfeather:
+
+QuickFeather
+############
+
+Overview
+********
+
+The QuickFeather development board is a platform with an on-board QuickLogic
+EOS S3 Sensor Processing Platform.
```

Comment:
It'd be nice to have a link to the product page.  https://www.quicklogic.com/products/eos-s3/quickfeather-development-kit/